q2_longitudinal/plugin_setup.py

```diff
@@ -221,3 +232,68 @@
         'composition. For more details and citation, please see '
         'doi.org/10.1002/gepi.22065')
 )
+
+
+plugin.methods.register_function(
+    function=first_differences,
+    inputs={'table': FeatureTable[RelativeFrequency]},
+    parameters={**miscellaneous_parameters,
+                **shared_parameters,
+                'metric': Str},
+    outputs=[('first_differences', SampleData[AlphaDiversity])],
+    input_descriptions={
+        'table': ('Feature table to optionally use for computing first '
+                  'differences.')},
+    parameter_descriptions={
+        **miscellaneous_parameter_descriptions,
+        **shared_parameter_descriptions,
+        'metric': ('Numerical metadata or artifact column to test. If '
+                   'computing first differences on a distance matrix, do not '
+                   'set a value for this parameter.'),
+    },
+    output_descriptions={'first_differences': 'Series of first differences.'},
+    name='First difference computation between sequential states',
+    description=(
+        'Calculates first differences in "metric" between sequential states '
+        'for samples collected from individual subjects sampled repeatedly at '
+        'two or more states. First differences can be performed on a metadata '
+        'column (including artifacts that can be input as metadata) or a '
+        'feature in a feature table. Outputs a data '
+        'series of first differences for each individual subject at each '
+        'sequential pair of states, labeled by the SampleID of the second '
+        'state (e.g., paired differences between time 0 and time 1 would be '
+        'labeled by the SampleIDs at time 1). This file can be used as input '
+        'to linear mixed effects models or other longitudinal or diversity '
+        'methods to compare changes in first differences across time or among '
+        'groups of subjects.')
+)
+
+
+plugin.methods.register_function(
+    function=first_distances,
+    inputs={'distance_matrix': DistanceMatrix},
+    parameters={**miscellaneous_parameters,
+                **shared_parameters},
+    outputs=[('first_distances', SampleData[AlphaDiversity])],
+    input_descriptions={
+        'distance_matrix': 'Matrix of distances between pairs of samples.'},
+    parameter_descriptions={
+        **miscellaneous_parameter_descriptions,
+        **shared_parameter_descriptions,
+    },
+    output_descriptions={'first_distances': 'Series of first distances.'},
+    name='First distance computation between sequential states',
+    description=(
+        'Calculates first distances between sequential states for samples '
+        'collected from individual subjects sampled repeatedly at two or more '
+        'states. This method is similar to the first differences method, '
```

Comment:
may be clearer to quote "first differences" or refer to it as `first_differences`

Reply:
👍